### PR TITLE
Added Get and List RPC

### PR DESCRIPTION
**Pull request by harshitap26**
Added the below RPCs:

NVMfRemoteControllerGet
NVMfRemoteControllerList

### CODE REVIEW

File: goopicsi.go [hunk 1]
```diff
@@ -126,6 +126,50 @@ func NVMeControllerConnect(request *pb.NVMfRemoteController) (*pb.NVMfRemoteCont
 	return &pb.NVMfRemoteControllerConnectResponse{}, nil
 }
 
+// NVMeControllerList lists all the connections to the remote NVMf controller
+func NVMeControllerList() (*pb.NVMfRemoteControllerListResponse, error) {
```
Comment (harishp8889) [inline goopicsi.go:130]: Please change this to some other `struct` instead of returning the proto messages

File: goopicsi.go [hunk 2]
```diff
@@ -126,6 +126,50 @@ func NVMeControllerConnect(request *pb.NVMfRemoteController) (*pb.NVMfRemoteCont
...
+	if err != nil {
+		log.Printf("could not list the connections to Remote NVMf controller: %v", err)
+		return nil, err
+	}
+	log.Printf("Connections: %v", response)
+	return response, nil
+}
+
+// NVMeControllerGet lists the connection to the remote NVMf controller corresponding to the given ID
+func NVMeControllerGet(request *pb.NVMfRemoteControllerGetRequest) (*pb.NVMfRemoteControllerGetResponse, error) {
```
Comment (harishp8889) [inline goopicsi.go:152]: Here as well, Avoid using the proto messages directly on the request and response objects.

**Review by harishp8889 — CHANGES_REQUESTED** [review]

File: goopicsi.go (hunk 2 above)
Comment (harshitap26) [reply goopicsi.go]: Fixed

File: goopicsi.go (hunk 1 above)
Comment (harshitap26) [reply goopicsi.go]: Fixed

**Review by harishp8889 — APPROVED** [review]

**Review by glimchb — APPROVED** [review]

File: goopicsi.go [hunk 3]
```diff
@@ -126,6 +126,50 @@ func NVMeControllerConnect(request *pb.NVMfRemoteController) (*pb.NVMfRemoteCont
...
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	response, err := client.NVMfRemoteControllerGet(ctx, &pb.NVMfRemoteControllerGetRequest{Id: id})
+	if err != nil {
+		log.Printf("could not list the connection to Remote NVMf controller corresponding to the given ID: %v", err)
+		return err
+	}
+	log.Printf("Connection corresponding to the given ID: %v", response)
+	return nil
```
Comment (glimchb) [inline goopicsi.go:170]: should you return some data here ?

**Pull request merged**